modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -313,6 +359,117 @@ Error GDScriptAnalyzer::resolve_inheritance(GDScriptParser::ClassNode *p_class,
 	class_type.native_type = result.native_type;
 	p_class->set_datatype(class_type);
 
+	// Check for invalid naming in members
+	GDScriptParser::DataType *current_data_type = &p_class->base_type;
+	while (current_data_type && current_data_type->kind == GDScriptParser::DataType::Kind::CLASS) {
```

Comment:
I just realized you're doing this loop in `resolve_inheritance()`, this check should not be done here (technically this should not fail the inheritance check itself, as long as the base class is valid). In `resolve_class_interface()` there's already a loop for the members and these checks can just be added there in the existing switch cases.

It also also allows for reporting multiple issues at once since it doesn't have to return at the first error.